### PR TITLE
feat(config): add consumption fee configs

### LINKED ISSUE #1541: Add configuration for the consumption fee
All parameters for calculating the fee are configurable. However, we recommend the following configurations:

For most validators, we suggest setting the unit price to 0 PAC and the fixed fee to 0.01 PAC. This configuration eliminates the consumption fee, meaning these validators will only apply the fixed fee.

For validators with good resources, we recommend setting the daily limit to 280 bytes, the unit price to 0.000005 PAC, and the fixed fee to 0 PAC. This allows users to send almost up to 5 free or low-fee transactions, but they will need to wait for these validators to enter the committee and propose the block. This way, the Pactus blockchain can support zero-fee transactions.

https://pips.pactus.org/PIPs/pip-31#:~:text=their%20preferred%20value.-,Configuration,-All%20parameters%20for

## Example config

```toml
# `tx_pool` contains configuration options for the transaction pool module.
[tx_pool]

  # `max_size` indicates the maximum number of unconfirmed transactions inside the pool.
  # Default is `1000`.
  max_size = 1000

  [tx_pool.fee]
    daily_limit = 280
    unit_price = 0.000005
    fixed_fee = 0.01
```

https://github.com/pactus-project/pactus/blob/2bcb8d60be6bef68509658dd71d967fce61d1090/config/example_config.toml#L122-L124

## Basic check

- `daily_limit` don't allowing zero value.

### PR DESCRIPTION
**Pull request by akbariandev**
## Description

In this PR new fee model (consumption fee) configs added.

## Related issue(s)

- Fixes #1541

### CODE REVIEW

File: config/example_config.toml [hunk 1]
```diff
@@ -119,9 +119,21 @@
...
+    # Default is `280` bytes.
+    daily_limit = 280
+
+    # `unit_price` indicates the fee of consumption unit.
+    # Default is `0.0` PAC.
+    unit_price = 0.0
+
+    # `fixed_price` is the constant fee applied to each transaction.
+    # Default is `0.01` PAC.
+    fixed_price = 0.01
```
Comment (b00f) [inline config/example_config.toml:136]: Please don't change the PIPs name. rename it to `fixed_fee`. 
Please move it to the top.

File: txpool/config.go [hunk 2]
```diff
@@ -5,14 +5,28 @@ import (
...
-	MaxSize   int     `toml:"max_size"`
-	MinFeePAC float64 `toml:"min_fee"`
+	MaxSize int       `toml:"max_size"`
+	Fee     FeeConfig `toml:"fee"`
+}
+
+type FeeConfig struct {
+	DailyLimit uint32  `toml:"daily_limit"`
+	UnitPrice  float64 `toml:"unit_price"`
+	FixedPrice float64 `toml:"fixed_price"`
```
Comment (b00f) [inline txpool/config.go:15]: Please rename it and move to the top.

File: txpool/config.go [hunk 3]
```diff
@@ -5,14 +5,28 @@ import (
...
+		MaxSize: 1000,
+		Fee:     DefaultFeeConfig(),
+	}
+}
+
+func DefaultFeeConfig() FeeConfig {
+	return FeeConfig{
+		DailyLimit: 280,
+		UnitPrice:  0,
+		FixedPrice: 0.01,
```
Comment (b00f) [inline txpool/config.go:29]: Same here

**Review by b00f — COMMENTED** [review]
I suggest this text for the example_config.toml

```toml
  # `tx_pool.fee` contains configuration to calulcate the transaction fee.
  [tx_pool.fee]

    # The `fixed_fee` is a constant fee applied to each transaction, regardless of its size or type.
    # Default is `0.01` PAC.
    fixed_fee = 0.01
   
    # The `daily_limit` is the number of bytes an account can send each day without paying a fee.
    # The `daily_limit` is part of he consumptional fee model.
    # To undesrstand how condumptional fee model works, you can refer to 
    # PIP-31: Consumptional Fee Mode: https://pips.pactus.org/PIPs/pip-31
    # Default is `280` bytes.
    daily_limit = 280

    # The `unit_price` defines the fee per byte in PAC.
    # The `unit_price` is part of he consumptional fee model.
    # To undesrstand how condumptional fee model works, you can refer to
    # PIP-31: Consumptional Fee Mode: https://pips.pactus.org/PIPs/pip-31
    # Default is `0.0` PAC.
    unit_price = 0.0
```

File: txpool/config.go [hunk 4]
```diff
@@ -5,14 +5,28 @@ import (
 )
 
 type Config struct {
-	MaxSize   int     `toml:"max_size"`
-	MinFeePAC float64 `toml:"min_fee"`
+	MaxSize int       `toml:"max_size"`
+	Fee     FeeConfig `toml:"fee"`
```
Comment (Ja7ad) [inline txpool/config.go:9]: @akbariandev Please update to

```go
Fee *FeeConfig `toml:"fee"`
```

File: txpool/config.go [hunk 5]
```diff
@@ -5,14 +5,28 @@ import (
...
 func DefaultConfig() *Config {
 	return &Config{
-		MaxSize:   1000,
-		MinFeePAC: 0.01,
+		MaxSize: 1000,
+		Fee:     DefaultFeeConfig(),
+	}
+}
+
+func DefaultFeeConfig() FeeConfig {
```
Comment (Ja7ad) [inline txpool/config.go:25]: @akbariandev Please change `FeeConfig` to `*FeeConfig`

File: txpool/config.go [hunk 6]
```diff
@@ -5,14 +5,28 @@ import (
...
+	return FeeConfig{
```
Comment (Ja7ad) [inline txpool/config.go:26]: @akbariandev Please change to `return *FeeConfig{}`

**Review by Ja7ad — CHANGES_REQUESTED** [review]

File: txpool/config.go [hunk 7]
```diff
@@ -26,7 +26,7 @@ func DefaultFeeConfig() FeeConfig {
 	return FeeConfig{
 		DailyLimit: 280,
 		UnitPrice:  0,
-		FixedPrice: 0.01,
+		FixedFee:   0.01,
```
Comment (b00f) [inline txpool/config.go:29]: Same here

File: config/example_config.toml [hunk 8]
```diff
@@ -119,9 +119,26 @@
...
+  [tx_pool.fee]
+
+    # The `fixed_fee` is a constant fee applied to each transaction, regardless of its size or type.
+    # Default is `0.01` PAC.
+    fixed_fee = 0.01
+
+    # The `daily_limit` is the number of bytes an account can send each day without paying a fee.
+    # The `daily_limit` is part of he consumptional fee model.
+    # To undesrstand how condumptional fee model works, you can refer to
+    # PIP-31: Consumptional Fee Mode: https://pips.pactus.org/PIPs/pip-31
```
Comment (b00f) [inline config/example_config.toml:132]:
```suggestion
    # PIP-31: Consumptional Fee Mode (https://pips.pactus.org/PIPs/pip-31)
```

File: config/example_config.toml [hunk 9]
```diff
@@ -119,9 +119,26 @@
...
+    # Default is `280` bytes.
+    daily_limit = 280
+
+    # The `unit_price` defines the fee per byte in PAC.
+    # The `unit_price` is part of he consumptional fee model.
+    # To undesrstand how condumptional fee model works, you can refer to
+    # PIP-31: Consumptional Fee Mode: https://pips.pactus.org/PIPs/pip-31
```
Comment (b00f) [inline config/example_config.toml:139]:
```suggestion
    # PIP-31: Consumptional Fee Mode (https://pips.pactus.org/PIPs/pip-31)
```

**Review by Ja7ad — APPROVED** [review]

**Review by b00f — APPROVED** [review]

**Pull request merged**